src/DataTypes/DataTypeFactory.h

```diff
@@ -35,6 +35,7 @@ class DataTypeFactory final : private boost::noncopyable, public IFactoryWithAli
     DataTypePtr get(const ASTPtr & ast) const;
     DataTypePtr getCustom(DataTypeCustomDescPtr customization) const;
     DataTypePtr getCustom(const String & base_name, DataTypeCustomDescPtr customization) const;
+    void setCustom(DataTypePtr & type, DataTypeCustomDescPtr customization) const;
```

Comment:
This is extremely confusing for 2 reasons:
* It's only using DataTypeFactory because it's a friend of IDataType.
* It could be static. It's not modifying and does not need the global DataTypeFactory instance.

Reply:
Alright, I agree it can be confusing. Fixed in [`8c0a5ad` (#77304)](https://github.com/ClickHouse/ClickHouse/pull/77304/commits/8c0a5add179432f65187e77497f14a2b3389e1b0)

But I guess the same could be said for `getCustom`->`get`->`getImpl`->`tryParseQuery`/`parseQuery`